packages/backend/src/dbt/dbtCliClient.ts

```diff
@@ -229,9 +243,17 @@ export class DbtCliClient implements DbtClient {
                 },
             },
             async () => {
-                const logs = await this._runDbtCommand(
-                    this.useDbtLs ? 'ls' : 'compile',
-                );
+                const dbtCommand = [];
+                const selector = this.selector?.trim();
+                if (selector) {
+                    if (!DbtCliClient.validateSelector(selector)) {
+                        throw new ParseError('Invalid dbt selector format');
+                    }
+                    dbtCommand.push('compile', '--select', `${selector}`);
```

Comment:
it seems we need `compile` when using selector,s otherwise the models will not be exactracted correctly from the manifest